Docker tests fail on tagged commits

It appears that we have [some Appveyor tests](https://ci.appveyor.com/project/joshmeek/prefect/builds/30468589) that fail on tagged released commits only because our Docker storage logic recognizes that the commit is tagged.  Full traceback here:

```
================================== FAILURES ===================================
_______ test_empty_docker_storage[win32-npipe:////./pipe/docker_engine] _______
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x0000003395A782C8>
platform = 'win32', url = 'npipe:////./pipe/docker_engine'
    @pytest.mark.parametrize(
        "platform,url",
        [
            ("win32", "npipe:////./pipe/docker_engine"),
            ("darwn", "unix://var/run/docker.sock"),
        ],
    )
    def test_empty_docker_storage(monkeypatch, platform, url):
        monkeypatch.setattr("prefect.environments.storage.docker.sys.platform", platform)
        monkeypatch.setattr(sys, "version_info", MagicMock(major=3, minor=6))
    
        storage = Docker()
    
        assert not storage.registry_url
>       assert storage.base_image == "python:3.6-slim"
E       AssertionError: assert 'prefecthq/prefect:0.9.2-python3.6' == 'python:3.6-slim'
E         - prefecthq/prefect:0.9.2-python3.6
E         + python:3.6-slim
tests\environments\storage\test_docker_storage.py:57: AssertionError
_________ test_empty_docker_storage[darwn-unix://var/run/docker.sock] _________
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x0000003395B55E88>
platform = 'darwn', url = 'unix://var/run/docker.sock'
    @pytest.mark.parametrize(
        "platform,url",
        [
            ("win32", "npipe:////./pipe/docker_engine"),
            ("darwn", "unix://var/run/docker.sock"),
        ],
    )
    def test_empty_docker_storage(monkeypatch, platform, url):
        monkeypatch.setattr("prefect.environments.storage.docker.sys.platform", platform)
        monkeypatch.setattr(sys, "version_info", MagicMock(major=3, minor=6))
    
        storage = Docker()
    
        assert not storage.registry_url
>       assert storage.base_image == "python:3.6-slim"
E       AssertionError: assert 'prefecthq/prefect:0.9.2-python3.6' == 'python:3.6-slim'
E         - prefecthq/prefect:0.9.2-python3.6
E         + python:3.6-slim
tests\environments\storage\test_docker_storage.py:57: AssertionError
_________ test_docker_init_responds_to_python_version[version_info0] __________
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x0000003395A78448>
version_info = (3, 5)
    @pytest.mark.parametrize("version_info", [(3, 5), (3, 6), (3, 7)])
    def test_docker_init_responds_to_python_version(monkeypatch, version_info):
        version_mock = MagicMock(major=version_info[0], minor=version_info[1])
        monkeypatch.setattr(sys, "version_info", version_mock)
        storage = Docker()
>       assert storage.base_image == "python:{}.{}-slim".format(*version_info)
E       AssertionError: assert 'prefecthq/prefect:0.9.2-python3.5' == 'python:3.5-slim'
E         - prefecthq/prefect:0.9.2-python3.5
E         + python:3.5-slim
tests\environments\storage\test_docker_storage.py:75: AssertionError
_________ test_docker_init_responds_to_python_version[version_info1] __________
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x0000003395CFF1C8>
version_info = (3, 6)
    @pytest.mark.parametrize("version_info", [(3, 5), (3, 6), (3, 7)])
    def test_docker_init_responds_to_python_version(monkeypatch, version_info):
        version_mock = MagicMock(major=version_info[0], minor=version_info[1])
        monkeypatch.setattr(sys, "version_info", version_mock)
        storage = Docker()
>       assert storage.base_image == "python:{}.{}-slim".format(*version_info)
E       AssertionError: assert 'prefecthq/prefect:0.9.2-python3.6' == 'python:3.6-slim'
E         - prefecthq/prefect:0.9.2-python3.6
E         + python:3.6-slim
tests\environments\storage\test_docker_storage.py:75: AssertionError
_________ test_docker_init_responds_to_python_version[version_info2] __________
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x0000003395B17DC8>
version_info = (3, 7)
    @pytest.mark.parametrize("version_info", [(3, 5), (3, 6), (3, 7)])
    def test_docker_init_responds_to_python_version(monkeypatch, version_info):
        version_mock = MagicMock(major=version_info[0], minor=version_info[1])
        monkeypatch.setattr(sys, "version_info", version_mock)
        storage = Docker()
>       assert storage.base_image == "python:{}.{}-slim".format(*version_info)
E       AssertionError: assert 'prefecthq/prefect:0.9.2-python3.7' == 'python:3.7-slim'
E         - prefecthq/prefect:0.9.2-python3.7
E         + python:3.7-slim
tests\environments\storage\test_docker_storage.py:75: AssertionError
```

--- a/tests/environments/storage/test_docker_storage.py
+++ b/tests/environments/storage/test_docker_storage.py
@@ -50,6 +50,7 @@ def test_add_flow_to_docker():
 def test_empty_docker_storage(monkeypatch, platform, url):
     monkeypatch.setattr("prefect.environments.storage.docker.sys.platform", platform)
     monkeypatch.setattr(sys, "version_info", MagicMock(major=3, minor=6))
+    monkeypatch.setattr(prefect, "__version__", "0.9.2+c2394823")
 
     storage = Docker()
 
@@ -67,10 +68,39 @@ def test_empty_docker_storage(monkeypatch, platform, url):
     assert not storage.local_image
 
 
+@pytest.mark.parametrize(
+    "platform,url",
+    [
+        ("win32", "npipe:////./pipe/docker_engine"),
+        ("darwn", "unix://var/run/docker.sock"),
+    ],
+)
+def test_empty_docker_storage_on_tagged_commit(monkeypatch, platform, url):
+    monkeypatch.setattr("prefect.environments.storage.docker.sys.platform", platform)
+    monkeypatch.setattr(sys, "version_info", MagicMock(major=3, minor=6))
+    monkeypatch.setattr(prefect, "__version__", "0.9.2")
+
+    storage = Docker()
+
+    assert not storage.registry_url
+    assert storage.base_image == "prefecthq/prefect:0.9.2-python3.6"
+    assert not storage.image_name
+    assert not storage.image_tag
+    assert storage.python_dependencies == ["wheel"]
+    assert storage.env_vars == {
+        "PREFECT__USER_CONFIG_PATH": "/root/.prefect/config.toml"
+    }
+    assert not storage.files
+    assert storage.prefect_version
+    assert storage.base_url == url
+    assert not storage.local_image
+
+
 @pytest.mark.parametrize("version_info", [(3, 5), (3, 6), (3, 7)])
 def test_docker_init_responds_to_python_version(monkeypatch, version_info):
     version_mock = MagicMock(major=version_info[0], minor=version_info[1])
     monkeypatch.setattr(sys, "version_info", version_mock)
+    monkeypatch.setattr(prefect, "__version__", "0.9.2+c2394823")
     storage = Docker()
     assert storage.base_image == "python:{}.{}-slim".format(*version_info)
 

